TrafficReplayer/src/org/opensearch/migrations/replay/RequestResponseResponseTriple.java

```diff
@@ -0,0 +1,84 @@
+package org.opensearch.migrations.replay;
+
+import com.google.common.primitives.Bytes;
+import org.json.HTTP;
+import org.json.JSONObject;
+import org.opensearch.migrations.replay.TrafficReplayer.RequestResponsePacketPair;
+
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+
+public class RequestResponseResponseTriple<write> {
+    private RequestResponsePacketPair sourcePair;
+    private List<byte[]> shadowResponseData;
+    Duration shadowResponseDuration;
+
+    public RequestResponseResponseTriple(RequestResponsePacketPair sourcePair,
+                                         List<byte[]> shadowResponseData,
+                                         Duration shadowResponseDuration) {
+        this.sourcePair = sourcePair;
+        this.shadowResponseData = shadowResponseData;
+        this.shadowResponseDuration = shadowResponseDuration;
+    }
+
+    private JSONObject jsonFromHttpData(List<byte[]> data) {
```

Comment:
I'd pull all of the JSON stuff into your writer class to separate responsibilities.